engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsNode.java

```diff
@@ -88,11 +84,6 @@ boolean equalsLongBool(long self, boolean other) {
     return false;
   }
 
-  @Specialization
-  boolean equalsLongInt(long self, int other) {
-    return self == (long) other;
```

Comment:
After a30ab0413 we have a failure in [Python interop test](https://github.com/enso-org/enso/actions/runs/4702123213/jobs/8339012848?pr=6280#step:10:10282). When comparing **Python array** and **Enso array** for `==` we use `interop.readArrayElement` without a conversion - as such we get `Integer` in instead of `Long` from Python and later `EqualsNode` decides that `Long(10) != Integer(10)`.

We have to review the usage of `InteropLibrary` in `Any.==` and make sure the obtained elements are subject to `HostValueToEnsoNode` conversion or use `ArrayAtNode`/`ArrayLengthNode` directly. CCing @Akirathan. Affects at least `equalsArrays`, `equalsHashMaps`, `equalsInteropObjectWithMembers`.

Reply:
Using `HostValueToEnsoNode` in 7c8730b and reporting #6299 for further consideration.